crates/block-producer/src/batch_builder/mod.rs

```diff
@@ -27,38 +29,35 @@ use crate::{
 /// of provers for proof generation. Proving is currently unimplemented and is instead simulated via
 /// the given proof time and failure rate.
 pub struct BatchBuilder {
-    pub batch_interval: Duration,
-    pub workers: NonZeroUsize,
-    /// Used to simulate batch proving by sleeping for a random duration selected from this range.
-    pub simulated_proof_time: Range<Duration>,
-    /// Simulated block failure rate as a percentage.
-    ///
-    /// Note: this _must_ be sign positive and less than 1.0.
-    pub failure_rate: f32,
+    worker_pool: JoinSet<()>,
```

Comment:
Related to the previous comment, it may be good to add a brief explanation (either here or as doc comments for the struct) of the pattern we use with the `worker_pool`.